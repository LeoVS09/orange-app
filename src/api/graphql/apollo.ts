import ApolloClient, {
  MutationOptions,
  FetchResult,
  OperationVariables,
  QueryOptions,
  ApolloQueryResult, ExecutionResult
} from "apollo-boost";

export interface APIClient {
  mutate<T, TVariables = OperationVariables>(options: MutationOptions<T, TVariables>): Promise<ExecutionResult<T>>

  query<T, TVariables = OperationVariables>(options: QueryOptions<TVariables>): Promise<ExecutionResult<T>>;
}

export function makeClient(uri: string): APIClient {
  const client = new ApolloClient({
    uri,
    credentials: "include"
  });

  return {
    mutate: options => new Promise((resolve, reject) => {
      // fix bug with apollo throw error
      try {
        client.mutate(options)
          // @ts-ignore
          .then(resolve)
          .catch(reject)

      } catch (e) {
        reject(e)
      }

    }),
    query: options => new Promise((resolve, reject) => {

      try {
        client.query(options)
          // @ts-ignore
          .then(resolve)
          .catch(reject)

      } catch (e) {
        reject(e)
      }
    })
  };
}
