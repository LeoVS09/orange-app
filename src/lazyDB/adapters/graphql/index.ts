import { AosSchema } from '@/abstractObjectSchema'
import { GenerateQueryOptions, generateQuery } from './generateQuery'
import { schemaToQueryFields } from './schemaToQuery'

export * from './types'
export * from './schemaToQuery'
export * from './generateQuery'

export interface GenerateQueryFromAosSchemaOptions extends Omit<GenerateQueryOptions, 'fields'> {
    /** fields which will be inside query */
    schema: AosSchema
  }

/** Generate query from AOS schema, for more info check generateQuery function */
export const generateQueryFromAosSchema = ({ operation, schema, variables }: GenerateQueryFromAosSchemaOptions) =>
  generateQuery({
    operation,
    fields: schemaToQueryFields(schema),
    variables
  })
