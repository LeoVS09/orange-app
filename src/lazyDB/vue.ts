import { PluginFunction } from 'vue'

export const LazyDBFilters: PluginFunction<any> = Vue => {
  Vue.prototype.$isReading = () => {
  }

  Vue.filter('isReading', (value: any, property?: string) => false)
  // const result = isReading(value, property)
  // console.log('is reading', result)
  // return result
}
