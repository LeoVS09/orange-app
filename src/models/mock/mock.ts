import {
   FullProblem,
   PartialProgramInput,
   PartialProgramOutput, ProblemReadState,
   ProblemStatus, ProblemTestingStatus,
   Tag,
} from "@/models/problems";
import {PartialUserProfile, UserType} from "@/models/user";
import {randomId} from "@/components/utils";
import {Test, TestStatus} from "@/models/tests";


export function defaultPartialProfile(): PartialUserProfile {
   return {
      id: randomId(),
      userId: randomId(),
      login: 'SomeUser',
      firstName: 'First',
      lastName: 'Last',
      type: UserType.CONTESTANT
   }
}

export function defaultProblem(): FullProblem {
   return {
      id: '',
      name: '',
      description: '',
      difficulty: 0,

      createdAt: new Date(),
      updatedAt: new Date(),
      publicationDate: new Date(),

      author: {
         ...defaultPartialProfile(),
         login: 'Author',
         type: UserType.TEACHER
      },

      tester: {
         ...defaultPartialProfile(),
         login: 'Tester',
         type: UserType.TEACHER
      },

      tags: [mockTag('some'), mockTag('tags')],

      limits: {
         time: 30000,
         memory: 2048
      },

      io: {
         input: mockInput('stdin'),
         output: mockOutput('stdout'),
      },

      status: ProblemStatus.Synced,
      readState: ProblemReadState.Full,
      testingStatus: ProblemTestingStatus.NotTested,

      tests: []
   }
}

export function defaultTest(): Test {
   return {
      id: "",
      index: 0,
      input: "",
      output: "",
      status: TestStatus.ForCreate,
      createdAt: new Date(),
      updatedAt: new Date(),
      isPublic: false
   }
}

export function mockTag(name: string): Tag {
   return {
      id: 'test-' + name,
      name
   }
}

export function mockTags(): Array<Tag> {
   return [
      'implementation',
      'math',
      'greedy',
      'brute force',
      'data structures',
      'constructive algorithms',
      'sorting',
      'binary search',
      'graphs',
      'trees',
      'strings',
      'number theory',
      'geometry',
      'combinatorics',
      'two pointers',
      'bitmasks',
      'probabilities',
      'shortest paths',
      'hashing',
      'divide and conquer',
      'games',
      'matrices',
      'flows',
      'string suffix structures',
      'expression parsing',
      'graph matchings',
      'ternary search',
      'meet-in-the-middle',
      '2-set',
      'chinese remainder theorem',
      'schedules'
   ].map(mockTag)
}

export function mockInput(name: string): PartialProgramInput {
   return {
      id: 'tests.ts' + name,
      name
   }
}

export function mockOutput(name: string): PartialProgramOutput {
   return {
      id: 'tests.ts' + name,
      name
   }
}
