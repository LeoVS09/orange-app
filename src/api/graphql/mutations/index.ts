import loginGql from './login.graphql'
import registerGql from './register.graphql'

import createProblemGql from './createProblem.graphql'
import updateProblemGql from './updateProblem.graphql'
import deleteProblemGql from './deleteProblem.graphql'

import createTestGql from './createTest.graphql'
import updateTestGql from './updateTest.graphql'
import deleteTestGql from './deleteTest.graphql'

import createTagGql from './createTag.graphql'
import updateTagGql from './updateTag.graphql'
import deleteTagGql from './deleteTag.graphql'

import createInputTypeGql from './createInputType.graphql'
import updateInputTypeGql from './updateInputType.graphql'
import deleteInputTypeGql from './deleteInputType.graphql'

import createOutputTypeGql from './createOutputType.graphql'
import updateOutputTypeGql from './updateOutputType.graphql'
import deleteOutputTypeGql from './deleteOutputType.graphql'

import gql from 'graphql-tag'
import {APIClient} from "../apollo";
import {
   ResponseLogin,
   ResponseRegister,
   RequestRegisterInput,
   RequestLoginInput,
   RequestUpdateProblemInput,
   ResponseUpdateProblem,
   RequestCreateProblemInput,
   ResponseCreateProblem,
   RequestDeleteProblemInput,
   ResponseDeleteProblem,
   RequestCreateTestInput,
   ResponseCreateTest,
   RequestUpdateTestInput,
   ResponseUpdateTest,
   RequestDeleteTestInput,
   ResponseDeleteTest,
   ResponseCreateTag,
   RequestCreateTagInput,
   RequestUpdateTagInput,
   ResponseUpdateTag,
   RequestDeleteTagInput,
   ResponseDeleteTag,
   ResponseUpdateInputType,
   ResponseCreateInputType,
   ResponseDeleteInputType,
   RequestCreateInputTypeInput,
   RequestUpdateInputTypeInput,
   RequestDeleteInputTypeInput,
   ResponseUpdateOutputType,
   RequestUpdateOutputTypeInput,
   ResponseCreateOutputType,
   RequestCreateOutputTypeInput,
   RequestDeleteOutputTypeInput,
   ResponseDeleteOutputType
} from "./types";
import {
   ResponseDataInputOutputType,
   ResponseDataProblem,
   ResponseDataTag,
   ResponseDataTest,
   ResponseDataUser
} from "../fragments/types";

// ------------------------------------------------ Auth ---------------------------------------------------------------

export const login = generateMutation<RequestLoginInput, ResponseLogin, ResponseDataUser>(
   loginGql,
      result => result.login.user
)

export const register = generateMutation<RequestRegisterInput, ResponseRegister, ResponseDataUser>(
   registerGql,
   result => result.register.user
)

// ----------------------------------------------- Problem -------------------------------------------------------------

export const createProblem = generateMutation<RequestCreateProblemInput, ResponseCreateProblem, ResponseDataProblem>(
   createProblemGql,
   result => result.createProblem.problem
)

export const updateProblem = generateMutation<RequestUpdateProblemInput, ResponseUpdateProblem, ResponseDataProblem>(
   updateProblemGql,
   result => result.updateProblem.problem
)

export const deleteProblem = generateMutation<RequestDeleteProblemInput, ResponseDeleteProblem, ResponseDataProblem>(
   deleteProblemGql,
   result => result.deleteProblem.problem
)

// ------------------------------------------------ Test ---------------------------------------------------------------

export const createTest = generateMutation<RequestCreateTestInput, ResponseCreateTest, ResponseDataTest>(
   createTestGql,
   result => result.createTest.test
)

export const updateTest = generateMutation<RequestUpdateTestInput, ResponseUpdateTest, ResponseDataTest>(
   updateTestGql,
   result => result.updateTest.test
)

export const deleteTest = generateMutation<RequestDeleteTestInput, ResponseDeleteTest, ResponseDataTest>(
   deleteTestGql,
   result => result.deleteTest.test
)

// ------------------------------------------------ Tag ----------------------------------------------------------------

export const createTag = generateMutation<RequestCreateTagInput, ResponseCreateTag, ResponseDataTag>(
   createTagGql,
   result => result.createTag.tag
)

export const updateTag = generateMutation<RequestUpdateTagInput, ResponseUpdateTag, ResponseDataTag>(
   updateTagGql,
   result => result.updateTag.tag
)

export const deleteTag = generateMutation<RequestDeleteTagInput, ResponseDeleteTag, ResponseDataTag>(
   deleteTagGql,
   result => result.deleteTag.tag
)

// --------------------------------------------- InputType -------------------------------------------------------------

export const createInputType = generateMutation<RequestCreateInputTypeInput, ResponseCreateInputType, ResponseDataInputOutputType>(
   createInputTypeGql,
   result => result.createProgramInputType.programInputType
)

export const updateInputType = generateMutation<RequestUpdateInputTypeInput, ResponseUpdateInputType, ResponseDataInputOutputType>(
   updateInputTypeGql,
   result => result.updateProgramInputType.programInputType
)

export const deleteInputType = generateMutation<RequestDeleteInputTypeInput, ResponseDeleteInputType, ResponseDataInputOutputType>(
   deleteInputTypeGql,
   result => result.deleteProgramInputType.programInputType
)

// --------------------------------------------- OutputType -------------------------------------------------------------

export const createOutputType = generateMutation<RequestCreateOutputTypeInput, ResponseCreateOutputType, ResponseDataInputOutputType>(
   createOutputTypeGql,
   result => result.createProgramOutputType.programOutputType
)

export const updateOutputType = generateMutation<RequestUpdateOutputTypeInput, ResponseUpdateOutputType, ResponseDataInputOutputType>(
   updateOutputTypeGql,
   result => result.updateProgramOutputType.programOutputType
)

export const deleteOutputType = generateMutation<RequestDeleteOutputTypeInput, ResponseDeleteOutputType, ResponseDataInputOutputType>(
   deleteOutputTypeGql,
   result => result.deleteProgramOutputType.programOutputType
)



function generateMutation<V, T, R>(graphql: any, formatter: (result: T) => R) {
   return (client: APIClient) => (variables: V) =>
      client.mutate<T>({
         mutation: gql(graphql),
         variables
      })
         .then(result => result.data && formatter(result.data))
}
