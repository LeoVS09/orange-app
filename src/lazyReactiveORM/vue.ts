import {PluginFunction} from "vue";
import {isReading} from "./ModelObserver";

export const LazyDBFilters: PluginFunction<any> = Vue => {
   Vue.prototype.$isReading = isReading

   Vue.filter('isReading', function (value: any, property?: string) {
      const result = isReading(value, property)
      console.log('is reading', result)
      return result
   })
}
