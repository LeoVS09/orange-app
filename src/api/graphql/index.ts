import * as mutations from './mutations'
import * as queries from './queries'
import {mapRequester} from "./utils";

// TODO: add pre-compiling queries when build project

export const login = mapRequester(mutations.login);

export const register = mapRequester(mutations.register);

export const currentUser = mapRequester(queries.currentUser);

export const searchCountries = mapRequester(queries.searchCountries);

export const countries = mapRequester(queries.countries);

export const problems = mapRequester(queries.problems)

export const problem = mapRequester(queries.problem)
