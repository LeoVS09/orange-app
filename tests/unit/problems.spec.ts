// @ts-ignore next-line
global.fetch = require('node-fetch')
import nock from 'nock'
import { createLocalVue, mount } from '@vue/test-utils'
import { VueConstructor } from 'vue'
import Vuex, {Store} from 'vuex'
import { nockAllGraphqlRequests } from './utils/graphql.mock'
import mockConsole from './utils/console.mock'
import flushPromises from 'flush-promises'
import vuexI18n from 'vuex-i18n'

const timeout = (time: number) => new Promise(resolve => setTimeout(resolve, time))

describe('Problems.vue', () => {

  let localVue: VueConstructor<any> 
  let store: Store<any> 

  beforeAll(() => nock.disableNetConnect());

  beforeEach(async () => {
    localVue = createLocalVue()
    localVue.use(Vuex)
    store = new Store({
      getters: {
        isTeacher: () => true
      }
    })

    localVue.use(vuexI18n.plugin, store)

    await nockAllGraphqlRequests(nock, 'problems')
  })

  afterEach(() => nock.cleanAll());

  it('renders problems list', async () => {
    const restoreConsole = mockConsole()
    // must be inside test for mock fetch function
    const Countries = require('@/pages/Problems.vue').default
    
    const wrapper = mount(Countries, { store, localVue, attachToDocument: true })
    
    // Wait while lazyDB requested all data and rerender
    await timeout(500)
    await flushPromises()

    expect(wrapper.element).toMatchSnapshot()
    restoreConsole()
  })
})
