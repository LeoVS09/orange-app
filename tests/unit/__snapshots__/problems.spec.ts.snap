// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Problems.vue renders problems list 1`] = `
<div
  class="problems"
>
  <div
    class="page-header highlight"
  >
    <div
      class="page-header--hidden-breadcrumbs"
    />
     
    <!---->
     
    <div
      class="page-header--content"
    >
      <h1>
        Problems
      </h1>
       
      <div
        class="page-header--actions"
      >
        <div
          class="button"
        >
          <button
            class="button--submit circle text-not-faded contrast text-on-hover icon-left"
          >
            <div
              class="button--content"
            >
              <span>
                <i
                  class="material-icons button--icon button--icon-left"
                >
                  add
                </i>
              </span>
               
              <span
                class="button--text"
              >
                Add problem
              </span>
               
              <!---->
            </div>
          </button>
        </div>
      </div>
       
      <!---->
    </div>
     
    <!---->
  </div>
   
  <div
    class="problems-list"
  >
    <div
      class="tags scrollable-horizontal"
    >
      <div
        class="tags--separator"
      >
        <span />
      </div>
       
      <span
        class="tags--left-gradient"
      />
       
      <div
        class="tags--content-wrapper"
        id="tags-xel7cv09rgq93o9ahoxy"
      >
        <div
          class="tags--content button-group horizontal secondary"
        >
          <div
            class="button-group--content"
          >
             
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    magni
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    id
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    dolore
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    sit
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    corrupti
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit simple text-not-faded secondary no-active-bold simple-active"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    officia
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
       
      <span
        class="tags--right-gradient"
      />
    </div>
     
    <section
      class="section"
    >
      <div
        class="filters"
      >
        <div
          class="button-group horizontal"
        >
          <div
            class="button-group--content"
          >
             
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit active gradient-highlight text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    All
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit gradient-highlight text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    Public
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit gradient-highlight text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    Not public
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button-group--item button"
            >
              <button
                class="button--submit gradient-highlight text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    Resolved
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="list--container"
      >
        <div
          class="list--hidden-columns"
        >
          <div
            class="list-column"
          >
            name
          </div>
           
          <div
            class="list-column"
          >
            difficulty
          </div>
           
          <div
            class="list-column"
          >
            author
          </div>
           
          <div
            class="list-column"
          >
            updated
          </div>
        </div>
         
        <div
          class="list"
        >
          <div
            class="list--header"
          >
            <div
              class="list--header-item"
            >
              <span
                class="list--header-text"
              >
                name
              </span>
               
              <div
                class="list--header-sort-arrow"
              >
                <!---->
              </div>
            </div>
            <div
              class="list--header-item"
            >
              <span
                class="list--header-text"
              >
                difficulty
              </span>
               
              <div
                class="list--header-sort-arrow"
              >
                <!---->
              </div>
            </div>
            <div
              class="list--header-item"
            >
              <span
                class="list--header-text"
              >
                author
              </span>
               
              <div
                class="list--header-sort-arrow"
              >
                <!---->
              </div>
            </div>
            <div
              class="list--header-item"
            >
              <span
                class="list--header-text"
              >
                updated
              </span>
               
              <div
                class="list--header-sort-arrow"
              >
                <!---->
              </div>
            </div>
          </div>
           
          <div
            class="list--add"
          >
            <div
              class="list--add-button button max-width max-height"
            >
              <button
                class="button--submit simple max-width max-height text-not-faded ghost icon-left left-align"
              >
                <div
                  class="button--content"
                >
                  <span>
                    <i
                      class="material-icons button--icon button--icon-left"
                    >
                      add
                    </i>
                  </span>
                   
                  <span
                    class="button--text"
                  >
                    Add
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
          </div>
           
          <div
            mode="out-in"
          >
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Optio ratione
                </span>
                <span
                  class="list-item--property"
                >
                  7
                </span>
                <span
                  class="list-item--property"
                >
                  Miss Lavada Hoeger
                </span>
                <span
                  class="list-item--property"
                >
                  15 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Molestias architecto aut
                </span>
                <span
                  class="list-item--property"
                >
                  23
                </span>
                <span
                  class="list-item--property"
                >
                  Mr. Wilma Corkery
                </span>
                <span
                  class="list-item--property"
                >
                  11 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Autem et aliquid
                </span>
                <span
                  class="list-item--property"
                >
                  14
                </span>
                <span
                  class="list-item--property"
                >
                  Ms. Maynard Abbott
                </span>
                <span
                  class="list-item--property"
                >
                  19 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Unde possimus
                </span>
                <span
                  class="list-item--property"
                >
                  8
                </span>
                <span
                  class="list-item--property"
                >
                  Mrs. Hardy Herzog
                </span>
                <span
                  class="list-item--property"
                >
                  15 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Enim non
                </span>
                <span
                  class="list-item--property"
                >
                  11
                </span>
                <span
                  class="list-item--property"
                >
                  Miss Furman Bernier
                </span>
                <span
                  class="list-item--property"
                >
                  7 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Nobis dolor
                </span>
                <span
                  class="list-item--property"
                >
                  14
                </span>
                <span
                  class="list-item--property"
                >
                  Mr. Stacey Bradtke
                </span>
                <span
                  class="list-item--property"
                >
                  13 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Necessitatibus temporibus enim
                </span>
                <span
                  class="list-item--property"
                >
                  11
                </span>
                <span
                  class="list-item--property"
                >
                  Mrs. Hildegard Jerde
                </span>
                <span
                  class="list-item--property"
                >
                  11 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Et architecto quia
                </span>
                <span
                  class="list-item--property"
                >
                  11
                </span>
                <span
                  class="list-item--property"
                >
                  Miss Karelle Feeney
                </span>
                <span
                  class="list-item--property"
                >
                  7 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Totam et
                </span>
                <span
                  class="list-item--property"
                >
                  21
                </span>
                <span
                  class="list-item--property"
                >
                  Mrs. Candelario Denesik
                </span>
                <span
                  class="list-item--property"
                >
                  6 days ago
                </span>
              </div>
            </div>
            <div
              class="list-item bordered hover-shadow"
              style="background: transparent;"
            >
              <div
                class="list-item--content"
              >
                <span
                  class="list-item--property"
                >
                  Suscipit dolor nihil
                </span>
                <span
                  class="list-item--property"
                >
                  5
                </span>
                <span
                  class="list-item--property"
                >
                  Mrs. Yasmine Skiles
                </span>
                <span
                  class="list-item--property"
                >
                  16 days ago
                </span>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="list--pagination"
        >
          <div
            class="button"
          >
            <button
              class="button--submit simple text-not-faded text-on-hover static-size icon-left"
            >
              <div
                class="button--content"
              >
                <span>
                  <i
                    class="material-icons button--icon button--icon-left"
                  >
                    navigate_before
                  </i>
                </span>
                 
                <span
                  class="button--text"
                >
                  Previous
                </span>
                 
                <!---->
              </div>
            </button>
          </div>
           
          <div
            class="list--pages"
          >
            <!---->
             
            <div
              class="button"
            >
              <button
                class="button--submit simple active text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    1
         
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
            <div
              class="button"
            >
              <button
                class="button--submit simple text-not-faded"
              >
                <div
                  class="button--content"
                >
                  <!---->
                   
                  <span
                    class="button--text"
                  >
                    2
         
                  </span>
                   
                  <!---->
                </div>
              </button>
            </div>
             
            <!---->
          </div>
           
          <div
            class="button"
          >
            <button
              class="button--submit simple text-not-faded text-on-hover static-size icon-right"
            >
              <div
                class="button--content"
              >
                <!---->
                 
                <span
                  class="button--text"
                >
                  Next
                </span>
                 
                <span>
                  <i
                    class="material-icons button--icon button--icon-right"
                  >
                    navigate_next
                  </i>
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
